src/composables/percentageTool.ts

```diff
@@ -0,0 +1,115 @@
+import { use } from 'echarts/core';
+import { ElementEvent } from 'echarts';
+import { useInputListener } from './inputListener';
+import { roundTimeframe } from '@/shared/timemath';
+import { GraphicComponent } from 'echarts/components';
+
+use([GraphicComponent]);
+
+export function usePercentageTool(chartRef, theme: string, timeframe_ms: number) {
+  const inputListener = useInputListener();
+
+  const color = theme === 'dark' ? 'white' : 'black';
+
+  const mousePos = ref({ x: 0, y: 0 });
+  const startPos = ref({ x: 0, y: 0 });
+  const drawLimitPerSecond = 144;
+  const canDraw = ref(true);
+  const active = ref(false);
+
+  function mouseMove(e: ElementEvent) {
+    mousePos.value.x = e.offsetX;
+    mousePos.value.y = e.offsetY;
+
+    if (!active.value || !canDraw.value) return;
+    draw(e.offsetX, e.offsetY);
+  }
+
+  function drawStart() {
+    active.value = true;
+    startPos.value = { ...mousePos.value };
+    chartRef.value?.setOption({
+      dataZoom: [{ disabled: true }],
+      graphic: [
+        {
+          type: 'line',
+          shape: {
+            x1: mousePos.value.x,
+            x2: mousePos.value.x,
+            y1: mousePos.value.y,
+            y2: mousePos.value.y,
+          },
+          style: {
+            stroke: color,
+          },
+        },
+        { type: 'text', z: 5 },
+      ],
+    });
+  }
+
+  function drawEnd() {
+    active.value = false;
+    chartRef.value?.setOption({
+      dataZoom: [{ disabled: false }],
+      graphic: [{ $action: 'remove' }, { $action: 'remove' }],
+    });
+  }
+
+  function draw(x: number, y: number) {
+    const startValues = chartRef.value?.convertFromPixel({ seriesIndex: 0 }, [
+      startPos.value.x,
+      startPos.value.y,
+    ]);
+    const endValues = chartRef.value?.convertFromPixel({ seriesIndex: 0 }, [x, y]);
+    const startPrice = Number(startValues[1]);
+    const endPrice = Number(endValues[1]);
+    const startTime = roundTimeframe(timeframe_ms, Number(startValues[0]));
+    const endTime = roundTimeframe(timeframe_ms, Number(endValues[0]));
+    const timeDiff = Math.abs(startTime - endTime) / timeframe_ms;
+    const pct = Math.abs(((startPrice - endPrice) / startPrice) * 100).toFixed(2);
+
+    chartRef.value?.setOption({
+      graphic: [
+        {
+          shape: {
+            x2: x,
```

Comment:
That would be neat, but I think as an option and not the default, maybe with also holding down CTRL snap to candle lows or highs or something like that? I agree with leaving it for the future though, for me most of the time it's not necessary, I just want a ballpark figure, but occasionally it would be convenient.

Reply:
I've implemented the "snapping" now - it's pretty straightforward once the time math is in place.

I also took the liberty to switch the implementation to a semi-transparent rectangle as i find this easier to see where exactly what started (it'll auto-give you a horizontal line on both top and bottom).

We can revert 3083bef4bfb45564c1e4881f7562549fd226ad12 if you don't like this.

![image](https://github.com/freqtrade/frequi/assets/5024695/f977390e-1220-43cc-bfef-214205e0efff)